src/OrchardCore.Modules/OrchardCore.Contents/Views/Items/ContentCreatedEvent.Fields.Edit.cshtml

```diff
@@ -0,0 +1,8 @@
+@using OrchardCore.Contents.Workflows.ViewModels
+@model ContentCreatedEventViewModel
+
+<fieldset class="form-group" asp-validation-class-for="SelectedContentTypeNames">
+    <label asp-for="SelectedContentTypeNames">@T["Content Types"] <span asp-validation-for="SelectedContentTypeNames"></span></label>
+    <select asp-for="SelectedContentTypeNames" class="form-control" asp-items="Model.AvailableContentTypes"></select>
+    <span class="hint">@T["Select any content types to filter on."]</span>
```

Comment:
We have a reusable View Component to display all content types and select them. Reusing it means we can improve it everywhere it's used. https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.Flows/Views/BagPartSettings.Edit.cshtml#L6

Reply:
I won't repeat this comment for every activity that deals with content types.

Reply:
Cool. OK if I make a tag helper for it, since it will now be used in even more places?

Reply:
Also, can we refactor the viewcomponent to take in an `IEnumerable<string>` of selected content types, rather than a `string[]`?

Reply:
Also, should we not copy the view component's view to a shared location, or is it intended to be copied into the module wanting to use that component? Right now, when I use the component, I get the following error:

![image](https://user-images.githubusercontent.com/938393/35042001-2267819c-fb87-11e7-99f3-77156106f998.png)

Reply:
Alternatively, if I update the viewcomponent's code as follows, it will work as well:

`return View("/.Modules/OrchardCore.ContentTypes/Views/Shared/Components/SelectContentTypes/Default.cshtml", model);`

The good thing about this is that we can keep the view as part of the module and not have to copy it to a shared location outside of the module.

Reply:
Curiously enough, the viewcomponent works fine for the BagPart.Edit view, so I must be missing something. Looking into it.

Reply:
Wait, no wonder it works fine for the BagPart.Settings - the module driving it is the ContentTypes module: *http://localhost:2918/Admin/**ContentTypes**/Page/ContentParts/BagPart/Edit*, so obviously the view engine will find it in the expected path.

So my previous comment applies: if we want to be able to use this VC from other modules, we'll need to provide the full path to the view:

`return View("/.Modules/OrchardCore.ContentTypes/Views/Shared/Components/SelectContentTypes/Default.cshtml", model);`

Or copy the view to a shared location, but not sure that's better.

Reply:
Done

Reply:
Don't do that. Let's ask @jtkech what he can do for this.

Reply:
I am pretty sure we just need to update a location expander for view components, like we do for views and razor pages.

Reply:
@sebastienros @sfmskywalker there is an open issue for this #1274. Indeed as sipke saw

> A shape view whose driver is defined in ModuleA can invoke a shared view component defined in ModuleB but only if it is in the context of a main controller defined in ModuleB, the one which also defines the view component. So, the view location expander can find the component view.

The workaround was to implement a custom `IViewLocationExpanderProvider`, but by knowing the module which provides shared view components. So, to implement it ou of the box, maybe

> We would need to resolve the component name from the requested view name / path, then maybe lookup for the registered `ViewComponentFeatures` to see which modules provide this component and maybe see which of these modules (ordered by deps) are enabled ...

But i can work on it if you think it is worth, maybe there is a simpler solution.